test/unit/header.js

```diff
@@ -0,0 +1,182 @@
+import React from 'react';
+import Adapt from 'core/js/adapt';
+import Header from 'core/templates/header';
+import console from 'console';
+import { act } from 'react-dom/test-utils';
+import { afterEach, beforeEach, describe, expect, it, jest } from '@jest/globals';
+import device from 'core/js/device';
+import { unmountComponentAtNode, render } from 'react-dom';
+
+beforeAll(async () => {
+  global.console = console;
+  // await import('core/js/app.js');
+  // await new Promise((resolve, reject) => {
+  //   const a = setInterval(() => {
+  //     if (!Adapt.get('_isStarted')) return;
+  //     clearInterval(a);
+  //     resolve(Promise.resolve());
+  //   }, 250);
+  // });
+});
+
+let container = null;
+beforeEach(() => {
+  jest.clearAllMocks();
+  // setup a DOM element as a render target
+  container = document.createElement('div');
+
+  document.body.appendChild(container);
+  Adapt.course = {
+    get: jest.fn().mockReturnValue(
+      {
+        _components: {
+          _text: {
+            ariaRegion: 'Text Component Aria Text'
+          }
+        },
+        _extensions: {
+          _glossary: {
+            ariaRegion: 'Glossary Extension Aria Text'
+          }
+        }
+      }
+    )
+  };
+
+});
+
+afterEach(() => {
+  // cleanup on exiting
+  unmountComponentAtNode(container);
+  container.remove();
+  container = null;
+});
+
+describe('Header Component State', () => {
+  it('should display the header body', () => {
+    const body = 'test body text';
+    const props = {
+      body
+    };
+    act(() => {
+      render(<Header {...props} />, container);
+    });
+    expect(container.textContent).toBe(body);
+  });
+
+  it('should display the header instructions', () => {
+    const instruction = 'test instructions';
+    const props = {
+      instruction
+    };
+    act(() => {
+      render(<Header {...props} />, container);
+    });
+    expect(container.textContent).toBe(instruction);
+  });
+
+  it('should not display a header when there is no body text, display title or sized instruction set', () => {
+    const _id = 'header';
+    const props = {
+      _id
+    };
+    act(() => {
+      render(<Header {...props} />, container);
+    });
+    expect(container.textContent).toBe('');
+  });
+
+  it('should display the component description when the title, display title or instruction is not specified ' +
+    'and the _isA11yComponentDescriptionEnabled is enabled and the aria region is set', () => {
+    const props = {
+      _isA11yComponentDescriptionEnabled: true,
+      _component: 'text'
+    };
+    act(() => {
+      render(<Header {...props} />, container);
+    });
+    const ariaRegion = 'Text Component Aria Text';
```

Comment:
Where do these defaults/fallbacks come form? Can we import them possibly?

Reply:
We could. I'm not sure if it's wiser to have defined, uniform test text, or if they should be made up for each test.

Reply:
Happy to leave as-is for now. Might be something we tighten up later 👍